core/arch/arm/arm.mk

```diff
@@ -164,6 +164,7 @@ ifneq ($(filter ta_arm32,$(ta-targets)),)
 CFG_ARM32_ta_arm32 := y
 arch-bits-ta_arm32 := 32
 ta_arm32-platform-cppflags += $(arm32-platform-cppflags)
+ta_arm32-platform-cppflags += -I$(out-dir)/core/include/generated/
```

Comment:
My intention here was to avoid duplication. 

> And indeed, $(out-dir)/core/include/generated/arm32_sysreg.h defines accessors to registers that can never be used by user space.

What harm do you see here? BTW, this would require creating a custom `arm32_user_sysreg.txt` for user-space.

> So, how about generating $(out-dir)/.../libutee/include/generated/arm32_user_sysreg.h from core/arch/arm/kernel/arm32_sysreg.txt?

It would require duplicating header generation logic from `core/arch/arm/arm.mk` to some user-space Makefile.

> Regardless, the arm32_[user_]sysreg.h file needs to be copied to the TA dev kit, too. Otherwise TAs cannot use it.

Do you foresee any case where TAs would directly access sysregs?

Reply:
> Do you foresee any case where TAs would directly access sysregs?

Yes, for instance quick-and-dirty timing of a specific code sequence with `CNTVCT`/`CNTFRQ`.

In fact we can choose to export the user sysregs to the TAs or not; I'm OK with both options. The problem with the current state of this PR is that AFAICT `arm_user_sysreg.h` wil get copied to the "TA dev kit" in `$(out-dir)/export_ta_arm32/include`, but `arm32_sysreg.h` will not.

Reply:
I think the best would be to supply and export a `arm32_user_sysreg.h` in TA dev kit, just as `arm64_user_sysreg.h` is supplied.

Reply:
@jenswi-linaro we're on the same page I think.

Reply:
Ok, will add support to export a `arm32_user_sysreg.h` in TA dev kit.

Reply:
The line we're commenting on here should also be removed since it shouldn't be needed any longer.

Reply:
It is needed during user libraries build since TA dev kit isn't created by that time.

Reply:
> Add a dependency on the generated arm32_user_sysreg.h instead then.

You mean we should generate custom `arm32_user_sysreg.h` for user-space?

> I don't think the libraries should have access to some special system headers.

`ftrace` code is part of user libraries only and currently its the main reason for which we need to include system headers in user-space.

Reply:
> You mean we should generate custom arm32_user_sysreg.h for user-space?

Yes, didn't we agree on that above?

> `ftrace` code is part of user libraries only and currently its the main reason for which we need to include system headers in user-space.

Yes, but that doesn't mean that we should let that code poke in core `.h` files. It's better to have a clear separation.
￼

Reply:
> Yes, didn't we agree on that above?

It seems there was some confusion earlier as if wen want to export arm32 kernel system header to user-space or if we need to generate arm32 explicitly for user-space. Now its clear to me that we are looking for latter option. Will implement it.

Reply:
Updated PR as per this discussion.